### PR TITLE
Allow specifying a default serialization value if marshaled object is None

### PR DESCRIPTION
**Pull request by Ch00k**
I'm not 100% sure this feature is at all needed but I came across a situation when I needed to return an empty object instead of the dictionary with all null keys for a marshaled object.
My example is the following:
```python
@marshal_with(IMDB_DATA, default={}, envelope='imdb_data')
def get(self, movie_id):
    movie = Movie.query.get(movie_id)
    return movie.imdb_data
```
`Movie.imdb_data` is an SQLAlchemy one-to-one relationship. When I query `/movies/1083/imdb-data` and the movie does not have IMDB data associated I'd rather return an empty JSON object than the object with all null keys.

Basically the behavior is the same as `default` kwarg for `flask_restful.fileds` classes.

### CODE REVIEW

File: flask_restful/__init__.py [hunk 1]
```diff
@@ -557,6 +559,8 @@ def marshal(data, fields, envelope=None):
     OrderedDict([('data', OrderedDict([('a', 100)]))])
 
     """
+    if data is None and default != 'all_null':
```
Comment (joshfriend) [inline flask_restful/__init__.py]: I'm not a fan of using a magic string to invoke a given behavior.

**Comment by joshfriend** [comment]
I think this behavior would be better implemented in the `fields.Nested` class if possible. I'd like to avoid adding too much complexity to the `marshal` function.
